### PR TITLE
feat(stdlib/sql): added SQL to() function

### PR DESCRIPTION
**Pull request by roshie548**
### Done checklist
- [ ] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: stdlib/sql/to.go [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+			valueStrings := make([]string, 0, l)
+			// valueArgs holds all the values to pass into the query
+			valueArgs := make([]interface{}, 0, l*len(cols))
+
+			if t.spec.Spec.DriverName != "sqlmock" {
+				q := fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (%s)", t.spec.Spec.Table, strings.Join(newSQLTableCols, ","))
+				_, err = t.tx.Exec(q)
+				if err != nil {
+					if rbErr := t.tx.Rollback(); err != nil {
+						return fmt.Errorf("\n%s\n%s", err, rbErr)
```
Comment (aanthony1243) [inline stdlib/sql/to.go]: new lines in errors cause us problems.   make this "transaction failed (%s) while recovering from %s"

File: stdlib/sql/to.go [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+			// valueStrings is an array of valuePlaceHolders, which will be joined later
+			valueStrings := make([]string, 0, l)
+			// valueArgs holds all the values to pass into the query
+			valueArgs := make([]interface{}, 0, l*len(cols))
+
+			if t.spec.Spec.DriverName != "sqlmock" {
+				q := fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (%s)", t.spec.Spec.Table, strings.Join(newSQLTableCols, ","))
+				_, err = t.tx.Exec(q)
+				if err != nil {
+					if rbErr := t.tx.Rollback(); err != nil {
```
Comment (aanthony1243) [inline stdlib/sql/to.go]: shouldn't this transaction roll back in Finish since we return an error?

File: stdlib/sql/to.go [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+		err := tbl.Do(func(er flux.ColReader) error {
+			l := er.Len()
+
+			// valueStrings is an array of valuePlaceHolders, which will be joined later
+			valueStrings := make([]string, 0, l)
+			// valueArgs holds all the values to pass into the query
+			valueArgs := make([]interface{}, 0, l*len(cols))
+
+			if t.spec.Spec.DriverName != "sqlmock" {
+				q := fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (%s)", t.spec.Spec.Table, strings.Join(newSQLTableCols, ","))
```
Comment (aanthony1243) [inline stdlib/sql/to.go]: is this what detects the schema change?

Comment (roshie548) [reply stdlib/sql/to.go]: No, it creates a table if the user names a destination table that doesn't exist

File: stdlib/sql/to.go [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+			txErr = t.tx.Rollback()
+		}
+		if txErr != nil {
+			panic(txErr)
+		}
+	}
+	t.d.Finish(err)
+}
+
+func ExecuteQueries(tx *sql.Tx, s *ToSQLOpSpec, colNames []string, valueStrings *[]string, valueArgs *[]interface{}, writer *Writer) error {
```
Comment (jlapacik) [inline stdlib/sql/to.go]: Is it possible to remove `writer` from this signature? It seems like it's only included for testing purposes?

File: stdlib/sql/to_test.go [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+					},
+					Data: [][]interface{}{
+						{execute.Time(11), "a", 2.0, "one"},
+						{execute.Time(21), "a", 2.0, "one"},
+						{execute.Time(21), "b", 1.0, "seven"},
+						{execute.Time(31), "a", 3.0, "nine"},
+						{execute.Time(41), "c", 4.0, "elevendyone"},
+					},
+				}},
+				Result: &fsql.Writer{
```
Comment (jlapacik) [inline stdlib/sql/to_test.go]: I would prefer the process method and the insert statements to be tested in isolation. Is that possible? Maybe it's too difficult?

File: stdlib/sql/to_test.go [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+						{execute.Time(11), "a", 2.0, "one"},
+						{execute.Time(21), "a", 2.0, "one"},
+						{execute.Time(21), "b", 1.0, "seven"},
+						{execute.Time(31), "a", 3.0, "nine"},
+						{execute.Time(41), "c", 4.0, "elevendyone"},
+					},
+				}},
+				ColumnNames: []string{"_time", "_measurement", "_value", "fred" },
+				ValueStrings: [][]string{{"(?,?,?,?)", "(?,?,?,?)", "(?,?,?,?)", "(?,?,?,?)", "(?,?,?,?)"}},
+				ValueArgs:[][]interface{}{{values.Time(int64(execute.Time(11))).Time(), "a", 2.0, "one", values.Time(int64(execute.Time(21))).Time(), "a", 2.0, "one", values.Time(int64(execute.Time(21))).Time(), "b", 1.0, "seven", values.Time(int64(execute.Time(31))).Time(), "a", 3.0, "nine", values.Time(int64(execute.Time(41))).Time(), "c", 4.0, "elevendyone"}},
```
Comment (jlapacik) [inline stdlib/sql/to_test.go]: Formatting these with a row on a new line might be easier to read.

**Pull request merged**